src/main/java/com/onelogin/saml/Response.java

```diff
@@ -26,146 +28,198 @@
 
 public class Response {
 
-	private Document xmlDoc;
-	private NodeList assertions;
+	/**
+	 * A DOMDocument class loaded from the SAML Response (Decrypted).
+	 */
+	private Document document;
+
 	private Element rootElement;
 	private final AccountSettings accountSettings;
-	private final Certificate certificate;
+	private String response;
 	private String currentUrl;
 	private StringBuffer error;
 
+	private String audienceUrl;
+
+	private static final Logger log = LoggerFactory.getLogger(Response.class);
+
 	public Response(AccountSettings accountSettings) throws CertificateException {
 		error = new StringBuffer();
 		this.accountSettings = accountSettings;		
-		certificate = new Certificate();
-		certificate.loadCertificate(this.accountSettings.getCertificate());
 	}
-	
+
 	public Response(AccountSettings accountSettings, String response) throws Exception {
-		this(accountSettings);
+		this(accountSettings);		
 		loadXmlFromBase64(response);
 	}
 
-	public void loadXmlFromBase64(String response) throws Exception {
+	public void loadXmlFromBase64(String responseStr) throws Exception {
 		Base64 base64 = new Base64();
-		byte[] decodedB = base64.decode(response);
-		String decodedS = new String(decodedB);
-		xmlDoc = Utils.loadXML(decodedS);
-		System.out.println("xmlDoc [ "+xmlDoc.getDocumentElement()+" ]");
+		byte[] decodedB = base64.decode(responseStr);
+		this.response = new String(decodedB);
+		this.document = Utils.loadXML(this.response);
+		if(this.document == null){
+
+		}
 	}
 
 	// isValid() function should be called to make basic security checks to responses.
-	public boolean isValid(){
+	public boolean isValid(String... requestId){
```

Comment:
mean that are optional like in php requestId = null, for now we are not supporting getting the requestId but the name method is ready to support it when we have a chance.